### ISSUE TITLE
cannot convert message.Data (variable of type interface{}) to string

### ISSUE THREAD

**Issue by kevin-shelaga** [Fallenstedt/twitter-stream#22]
Consuming tweet example is incorrect:

```go
go func() {
		for message := range api.Stream.GetMessages() {
			if message.Err != nil {
				panic(message.Err)
			}
                        // Will print something like: 
                        //{"data":{"id":"1356479201000","text":"Look at this cat picture"},"matching_rules":[{"id":12345,"tag":"cat tweets with images"}]}
			fmt.Println(string(message.Data))    <----- WRONG
		}
	}()

	time.Sleep(time.Second * 30)
	api.Stream.StopStream()
```

Results in compiler error:

cannot convert message.Data (variable of type interface{}) to string

**Comment by Fallenstedt** [comment]
Ah, looks like the source of the error occurs when I attempt to unmarshal the MatchingRules ID. 
![Screen Shot 2021-11-24 at 4 43 53 PM](https://user-images.githubusercontent.com/13971824/143331969-0406bfac-1172-4e4a-97e1-330d819155a5.png)


I encountered this a while ago and mentioned it to the twitter devs too: https://twittercommunity.com/t/filtered-stream-matching-rules-id-is-now-a-string/159527

I'll update this example, and I will also create a new example which shows how a stream can be restarted when Twitter disconnects.

**Comment by kevin-shelaga** [comment]
Thanks, I found the following works. 

`fmt.Sprintf("%s", message.Data)`

**Comment by Fallenstedt** [comment]
I updated the examples a bit. I updated the example struct, and I also created a new example `restart_stream_example`. It shows how to handle disconnects from twitter gracefully. https://github.com/Fallenstedt/twitter-stream/blob/6f9abb8bdda4cbb626210ea7bc7bc0c70559c4ac/example/restart_stream_example.go

This will be available in 0.3.3 https://github.com/Fallenstedt/twitter-stream/pull/25
